### DISCUSSION TITLE
How to access parse_state from dsl::scan or lexy::rule_scanner?

### DISCUSSION THREAD

**Discussion by Korsar13** [foonathan/lexy#150] [Q&A] [answered]
```
struct known_user_type: lexy::scan_production<std::string>
{
	template <typename Reader, typename Context>
	static constexpr scan_result scan( lexy::rule_scanner<Context, Reader>& scanner )
	{
		// scanner.context() -- private method?
	}
};
```

**Comment by foonathan** [comment, marked as answer]
You don't want to access the context directly, it is an implementation detail. Actual parse state can be accessed in a second parameter. an example is here: https://github.com/foonathan/clauf/blob/25b91226c1e8193c4498149cbedf69c2068458da/src/compiler.cpp#L1458
